inference-engine/src/inference_engine/include/openvino/runtime/infer_request.hpp

```diff
@@ -0,0 +1,231 @@
+     */
+    enum WaitMode : int64_t {
+        /** Wait until inference result becomes available */
+        RESULT_READY = -1,
+        /** IInferRequest doesn't block or interrupt current thread and immediately returns inference status */
+        STATUS_ONLY = 0,
+    };
+
+    /**
+     * @brief Default constructor
+     */
+    InferRequest() = default;
+
+    /**
+     * @brief Sets input/output data to infer
+     *
+     * @note Memory allocation does not happen
+     * @param name Name of input or output blob.
+     * @param data Reference to input or output blob. The type of a blob must match the network input precision and
+     * size.
+     */
+    void set_blob(const std::string& name, const std::shared_ptr<InferenceEngine::Blob>& data);
+
+    /**
+     * @brief Gets input/output data for inference
+     *
+     * @note Memory allocation does not happen
+     * @param name A name of Blob to get
+     * @return A shared pointer to a Blob with a name @p name. If a blob is not found, an exception is thrown.
+     */
+    std::shared_ptr<InferenceEngine::Blob> get_blob(const std::string& name);
+
+    /**
+     * @brief Sets blob with a pre-process information
+     * @note Returns an error in case if data blob is output
+     * @param name Name of input blob.
+     * @param data A reference to input. The type of Blob must correspond to the network input precision and size.
+     * @param info Preprocess info for blob.
+     */
+    void set_blob(const std::string& name,
+                  const std::shared_ptr<InferenceEngine::Blob>& data,
+                  const InferenceEngine::PreProcessInfo& info);
+
+    /**
+     * @brief Gets pre-process for input data
+     * @param name Name of input blob.
+     * @return pointer to pre-process info of blob with name
+     */
+    const InferenceEngine::PreProcessInfo& get_pre_proccess(const std::string& name) const;
+
+    /**
+     * @brief Infers specified input(s) in synchronous mode
+     *
+     * @note blocks all methods of InferRequest while request is ongoing (running or waiting in queue)
+     *
+     */
+    void infer();
+
+    /**
+     * @brief Cancels inference request
+     */
+    void cancel();
+
+    /**
+     * @brief Queries performance measures per layer to get feedback of what is the most time consuming layer
+     *
+     * @note not all plugins provide meaningful data
+     * @return Map of layer names to profiling information for that layer
+     */
+    std::map<std::string, InferenceEngine::InferenceEngineProfileInfo> get_performance_counts() const;
+
+    /**
+     * @brief Sets input data to infer
+     *
+     * @note Memory allocation doesn't happen
+     * @param inputs A reference to a map of input blobs accessed by input names.
+     *        The type of Blob must correspond to the network input precision and size.
+     */
+    void set_input(const std::map<std::string, std::shared_ptr<InferenceEngine::Blob>>& inputs);
```

Comment:
It Will be changed when `Tensor` API is introduced

Reply:
The idea was to introduce new methods which use Blob instead of Tensor. It allows to align the new API with the proposal.
Can we do it? Or it will be better to do in the separate PR?

Reply:
The API should be like `Blob::Ptr get_tensor()`?
 renamed to `get_set_tensor` but operate with `ie::Blob::Ptr` ?

Reply:
Will be represented in separate PR